pkg/operator/ceph/cluster/watcher.go

```diff
@@ -201,47 +201,79 @@ func (c *clientCluster) fenceNode(ctx context.Context, node *corev1.Node, cluste
 
-	rbdVolumesInUse := getCephVolumesInUse(cluster, volumesInuse)
-	if len(rbdVolumesInUse) == 0 {
-		logger.Debugf("no rbd volumes in use for out of service node %q", node.Name)
+	rbdVolumesInUse, cephFSVolumeInUse := getCephVolumesInUse(cluster, volumesInuse)
+	if len(rbdVolumesInUse) == 0 && len(cephFSVolumeInUse) == 0 {
+		logger.Debugf("no rbd or cephFS volumes in use for out of service node %q", node.Name)
 		return nil
 	}
 
-	logger.Infof("node %q require fencing, found rbd volumes in use", node.Name)
 	listPVs, err := c.context.Clientset.CoreV1().PersistentVolumes().List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return pkgerror.Wrapf(err, "failed to list PV")
 	}
 
-	rbdPVList := listRBDPV(listPVs, cluster, rbdVolumesInUse)
-	if len(rbdPVList) == 0 {
-		logger.Debug("No rbd PVs found on the node")
-		return nil
-	}
+	if len(rbdVolumesInUse) != 0 {
+		rbdPVList := listRBDPV(listPVs, cluster, rbdVolumesInUse)
+		if len(rbdPVList) == 0 {
+			logger.Debug("No rbd PVs found on the node")
+		} else {
+			logger.Infof("node %q require fencing, found rbd volumes in use", node.Name)
+			clusterInfo, _, _, err := opcontroller.LoadClusterInfo(c.context, ctx, cluster.Namespace, &cluster.Spec)
+			if err != nil {
+				return pkgerror.Wrapf(err, "Failed to load cluster info.")
+			}
 
-	clusterInfo, _, _, err := opcontroller.LoadClusterInfo(c.context, ctx, cluster.Namespace, &cluster.Spec)
-	if err != nil {
-		return pkgerror.Wrapf(err, "Failed to load cluster info.")
+			for i := range rbdPVList {
+				err = c.fenceRbdImage(ctx, node, cluster, clusterInfo, rbdPVList[i])
+				// We only need to create the network fence for any one of rbd pv.
+				if err == nil {
+					break
+				}
+
+				if i == len(rbdPVList)-1 {
+					return pkgerror.Wrapf(err, "failed to fence rbd volumes")
+				}
+				logger.Errorf("failed to fence rbd volumes %q, trying next rbd volume", rbdPVList[i].Name)
+			}
+		}
 	}
 
-	for i := range rbdPVList {
-		err = c.fenceRbdImage(ctx, node, cluster, clusterInfo, rbdPVList[i])
-		// We only need to create the network fence for any one of rbd pv.
-		if err == nil {
-			break
+	if len(cephFSVolumeInUse) != 0 {
+		cephFSVolumeInUseMap := make(map[string]struct{})
+		for _, vol := range cephFSVolumeInUse {
+			cephFSVolumeInUseMap[vol] = struct{}{}
 		}
+		cephFSPVList := listRWOCephFSPV(listPVs, cluster, cephFSVolumeInUseMap)
+		if len(cephFSPVList) == 0 {
+			logger.Debug("No cephFS PVs found on the node")
+			return nil
+		}
+		logger.Infof("node %q require fencing, found cephFS volumes in use", node.Name)
+		clusterInfo, _, _, err := opcontroller.LoadClusterInfo(c.context, ctx, cluster.Namespace, &cluster.Spec)
+		if err != nil {
+			return pkgerror.Wrapf(err, "Failed to load cluster info.")
+		}
+
+		for i := range cephFSPVList {
+			err = c.fenceCephFSVolume(ctx, node, cluster, clusterInfo, cephFSPVList[i])
+			// We only need to create the network fence for any one of cephFS pv.
+			if err == nil {
```

Comment:
hi @riya-singhal31, why do you only need to set up network fencing for one pv? If I understand correctly, each pv may correspond to a different client ip. Don’t we need to block the client IP of each pv? Thanks.

Reply:
As far as I got to know from @Rakshith-R,  We block the nodePlugin ip so doing per node works.

Reply:
Thanks for your explaination, The ip addresses are obtained through the `ceph tell mds.* client ls` or `rbd status`. It seems not to be the nodePlugin ip. Not sure if I have misunderstandings.

Reply:
I'll discuss with CSI/csi-addons team how it works in the backend after the cool-down time and update here

Reply:
> Thanks for your explaination, The ip addresses are obtained through the `ceph tell mds.* client ls` or `rbd status`. It seems not to be the nodePlugin ip. Not sure if I have misunderstandings.

It is the nodeplugin or in some cases holder pod IP.
All pods using rbd or cephfs are mounted from the same rbd/cephfs nodeplugin(or holder pod network ns), therefore, from Ceph perpective share the same IP. 

You can mount volumes and check their IP yourself from ceph's perspective.

Reply:
There exists a scenario where users manually mount rbd or cephfs instead of using it inside the storage cluster. Client IP will be different from nodePlugin ip.

Reply:
> There exists a scenario where users manually mount rbd or cephfs instead of using it inside the storage cluster. Client IP will be different from nodePlugin ip.

This is designed for fencing rbd or cephfs PVCs inside a faulty kubernetes node only.

Reply:
Got it, thanks for your nice explaination.